frontend/webapp/reuseable-components/data-card/data-card-fields/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { Fragment } from 'react';
+import styled from 'styled-components';
+import { DataTab, Divider, InstrumentStatus, MonitorsIcons, Status, Text, Tooltip } from '@/reuseable-components';
+import { capitalizeFirstLetter, getProgrammingLanguageIcon, parseJsonStringToPrettyString, safeJsonParse, WORKLOAD_PROGRAMMING_LANGUAGES } from '@/utils';
+
+export interface DataCardRow {
+  type?: 'divider' | 'break-line' | 'monitors' | 'active-status' | 'source-container';
+  title?: string;
+  tooltip?: string;
+  value?: string;
+}
+
+interface Props {
+  data: DataCardRow[];
+}
+
+const ListContainer = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  gap: 16px 32px;
+  width: 100%;
+`;
+
+const ListItem = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 2px;
+`;
+
+const ItemTitle = styled(Text)`
+  color: ${({ theme }) => theme.text.grey};
+  font-size: 10px;
+  line-height: 16px;
+`;
+
+const ItemValue = styled(Text)`
+  color: ${({ theme }) => theme.colors.text};
+  font-size: 12px;
+  line-height: 18px;
+`;
+
+const STRETCH_TYPES = ['source-container']; // Types that should stretch to 100% width
+
+export const DataCardFields: React.FC<Props> = ({ data }) => {
+  return (
+    <ListContainer>
+      {data.map(({ type, title, tooltip, value }) => {
+        return (
+          <ListItem key={`card-data-${type}-${title}-${value}`} style={{ width: !!type && STRETCH_TYPES.includes(type) ? '100%' : 'unset' }}>
```

Comment:
```suggestion
          <ListItem key={`card-data-${type}-${title}-${value}`} style={{ width: !!type && STRETCH_TYPES.includes(type) ? '100%' : 'unset' }}>
```
we can now use  react useId instead of generate our own id